source/configure/calls-troubleshooting.md

```diff
@@ -0,0 +1,385 @@
+
+1. **Enable debug logging** in System Console > Environment > Logging > File Log Level
+
+2. **Filter for Calls-related logs**:
+
+   ```bash
+   grep -i "calls" /path/to/mattermost.log
+   ```
+      
+3. **Look for common patterns**:
+   
+   - Connection errors to RTCD
+   - Plugin initialization issues
+   - WebSocket connection problems
+
+### Browser Console Logs
+
+Instruct users to check their browser console logs:
+
+1. **In Chrome/Edge**:
+   - Press F12 to open Developer Tools
+   - Go to the Console tab
+   - Look for errors related to WebRTC, Calls, or media permissions
+
+2. **Specific patterns to look for**:
+   
+   - "getUserMedia" errors (microphone permission issues)
+   - "ICE connection" failures
+   - WebSocket connection errors
+
+## Performance Issues
+
+### Diagnosing High CPU Usage
+
+If RTCD servers show high CPU usage:
+
+1. **Check concurrent calls and participants**:
+   
+   - Access the Prometheus metrics endpoint to see active sessions
+   - Compare with the benchmark data in the {doc}`Calls Metrics and Monitoring <calls-metrics-monitoring>` documentation's Performance Baselines section
+
+2. **Profile CPU usage** (Linux):
+
+   ```bash
+   top -p $(pgrep rtcd)
+   ```
+      
+   Or for detailed per-thread usage:
+   
+   ```bash
+   ps -eLo pid,ppid,tid,pcpu,comm | grep rtcd
+   ```
+
+3. **Enable pprof profiling** (if needed):
+
+   Add to your RTCD configuration:
+   
+   ```json
+   {
+     "debug": {
+       "pprof": true,
+       "pprofPort": 6060
+     }
+   }
+   ```
+      
+   Then capture a CPU profile:
+   
+   ```bash
+   curl http://localhost:6060/debug/pprof/profile > cpu.profile
+   ```
+      
+   Analyze with:
+   
+   ```bash
+   go tool pprof -http=:8080 cpu.profile
+   ```
+
+### Diagnosing Network Bottlenecks
+
+If you suspect network bandwidth issues:
+
+1. **Monitor network utilization**:
+
+   ```bash
+   iftop -n
+   ```
+      
+2. **Check for packet drops**:
+
+   ```bash
+   netstat -su | grep -E 'drop|error'
+   ```
+      
+3. **Verify system network buffers**:
+
+   ```bash
+   sysctl -a | grep net.core.rmem
+   sysctl -a | grep net.core.wmem
+   ```
+      
+   Ensure these match the recommended values:
+   
+   ```bash
+   net.core.rmem_max = 16777216
+   net.core.wmem_max = 16777216
+   net.core.optmem_max = 16777216
+   ```
+
+## Recording and Transcription Issues
+
+For troubleshooting calls-offloader service issues including recording and transcription problems, see the [Calls Offloader Setup and Configuration](calls-offloader-setup.md#troubleshooting) guide.
+
+### Calls-Offloader Docker Debugging
+
+If you're running calls-offloader in Docker, use these commands for debugging:
+
+#### Monitor Live Logs
+
+To view real-time logs from calls-offloader containers:
+
+```bash
+# Find and follow logs from all calls-related containers
+docker ps --format "{{.ID}} {{.Image}}" | grep "calls" | awk '{print $1}' | xargs -I {} docker logs -f {}
+```
```

Comment:
Similar to a previous comment, using the [dedicated config setting](https://github.com/mattermost/calls-offloader/blob/8ddde65d2088490e6dab9a4d844115d35e67f7c9/config/config.sample.toml#L56-L59) seems simpler.